From libc-ports-return-1327-listarch-libc-ports=sources dot redhat dot com at sourceware dot org Tue Aug 25 12:52:22 2009
Return-Path: <libc-ports-return-1327-listarch-libc-ports=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-libc-ports at sources dot redhat dot com
Received: (qmail 29976 invoked by alias); 25 Aug 2009 12:52:21 -0000
Received: (qmail 29965 invoked by uid 22791); 25 Aug 2009 12:52:20 -0000
X-SWARE-Spam-Status: No, hits=-1.9 required=5.0 	tests=AWL,BAYES_00,SPF_PASS
X-Spam-Check-By: sourceware.org
Received: from mail.codesourcery.com (HELO mail.codesourcery.com) (65.74.133.4)     by sourceware.org (qpsmtpd/0.43rc1) with ESMTP; Tue, 25 Aug 2009 12:52:14 +0000
Received: (qmail 25083 invoked from network); 25 Aug 2009 12:52:12 -0000
Received: from unknown (HELO mbp.local) (maxim@127.0.0.2)   by mail.codesourcery.com with ESMTPA; 25 Aug 2009 12:52:12 -0000
Message-ID: <4A93DE74.1070103@codesourcery.com>
Date: Tue, 25 Aug 2009 16:52:04 +0400
From: Maxim Kuvyrkov <maxim at codesourcery dot com>
User-Agent: Thunderbird 2.0.0.23 (Macintosh/20090812)
MIME-Version: 1.0
To: libc-ports at sourceware dot org
CC: Andreas Schwab <schwab at linux-m68k dot org>
Subject: [M68K/ColdFire patch 3/n] Update jmpbuf-unwind.h
References: <4A93D885.3060004@codesourcery.com>
In-Reply-To: <4A93D885 dot 3060004 at codesourcery dot com>
Content-Type: multipart/mixed;  boundary="------------010909010302070202070003"
X-IsSubscribed: yes
Mailing-List: contact libc-ports-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Id: <libc-ports.sourceware.org>
List-Subscribe: <mailto:libc-ports-subscribe at sourceware dot org>
List-Post: <mailto:libc-ports at sourceware dot org>
List-Help: <mailto:libc-ports-help at sourceware dot org>, <http://sourceware dot org/lists dot html#faqs>
Sender: libc-ports-owner at sourceware dot org
Delivered-To: mailing list libc-ports at sourceware dot org

This is a multi-part message in MIME format.
--------------010909010302070202070003
Content-Type: text/plain; charset=ISO-8859-1; format=flowed
Content-Transfer-Encoding: 7bit

This patch updates jmpbuf-unwind.h.

The implementation was borrowed from what other major architectures do.

Any comments?

Thanks,

--
Maxim K.
CodeSourcery

--------------010909010302070202070003
Content-Type: text/plain; x-mac-type="0"; x-mac-creator="0";
 name="0003-Update-jmpbuf-unwind.h.patch"
Content-Transfer-Encoding: 7bit
Content-Disposition: inline;
 filename="0003-Update-jmpbuf-unwind.h.patch"

>From efd28b55921422dc141672336bed8a33284dfbfb Mon Sep 17 00:00:00 2001
From: Maxim Kuvyrkov <maxim@codesourcery.com>
Date: Mon, 24 Aug 2009 12:35:48 +0400
Subject: [PATCH 3/5] Update jmpbuf-unwind.h.

	* sysdeps/m68k/jmpbuf-unwind.h: Update.
---
 sysdeps/m68k/jmpbuf-unwind.h |   23 ++++++++++++++++++++++-
 1 files changed, 22 insertions(+), 1 deletions(-)

diff --git a/sysdeps/m68k/jmpbuf-unwind.h b/sysdeps/m68k/jmpbuf-unwind.h
index 3490c79..d001039 100644
--- a/ports/sysdeps/m68k/jmpbuf-unwind.h
+++ b/ports/sysdeps/m68k/jmpbuf-unwind.h
@@ -1,5 +1,5 @@
 /* Examine __jmp_buf for unwinding frames.  m68k version.
-   Copyright (C) 2006 Free Software Foundation, Inc.
+   Copyright (C) 2006, 2009 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -18,8 +18,29 @@
    02111-1307 USA.  */
 
 #include <setjmp.h>
+#include <stdint.h>
+#include <unwind.h>
 
 /* Test if longjmp to JMPBUF would unwind the frame
    containing a local variable at ADDRESS.  */
 #define _JMPBUF_UNWINDS(jmpbuf, address, demangle)		\
   ((void *) (address) < (void *) demangle ((jmpbuf)->__sp))
+
+#define _JMPBUF_CFA_UNWINDS_ADJ(_jmpbuf, _context, _adj) \
+  _JMPBUF_UNWINDS_ADJ (_jmpbuf, (void *) _Unwind_GetCFA (_context), _adj)
+
+static inline uintptr_t __attribute__ ((unused))
+_jmpbuf_sp (__jmp_buf regs)
+{
+  uintptr_t sp = regs[0].__sp;
+#ifdef PTR_DEMANGLE
+  PTR_DEMANGLE (sp);
+#endif
+  return sp;
+}
+
+#define _JMPBUF_UNWINDS_ADJ(_jmpbuf, _address, _adj) \
+  ((uintptr_t) (_address) - (_adj) < _jmpbuf_sp (_jmpbuf) - (_adj))
+
+/* We use the normal longjmp for unwinding.  */
+#define __libc_unwind_longjmp(buf, val) __libc_longjmp (buf, val)
-- 
1.6.4


--------------010909010302070202070003--

